src/main/java/uk/gov/hmcts/juror/api/moj/report/grouped/TrialAttendanceReport.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.juror.api.moj.report.grouped;
+
+import com.querydsl.core.Tuple;
+import com.querydsl.jpa.impl.JPAQuery;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.juror.api.moj.controller.reports.request.StandardReportRequest;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.GroupedReportResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.GroupedTableData;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.StandardReportResponse;
+import uk.gov.hmcts.juror.api.moj.domain.QReportsJurorPayments;
+import uk.gov.hmcts.juror.api.moj.domain.trial.Trial;
+import uk.gov.hmcts.juror.api.moj.report.AbstractGroupedReport;
+import uk.gov.hmcts.juror.api.moj.report.ReportGroupBy;
+import uk.gov.hmcts.juror.api.moj.report.datatypes.ReportsJurorPaymentsDataTypes;
+import uk.gov.hmcts.juror.api.moj.repository.trial.TrialRepository;
+import uk.gov.hmcts.juror.api.moj.utils.SecurityUtil;
+
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class TrialAttendanceReport extends AbstractGroupedReport {
+
+    private final TrialRepository trialRepository;
+
+    @Autowired
+    public TrialAttendanceReport(TrialRepository trialRepository) {
+        super(QReportsJurorPayments.reportsJurorPayments,
+            ReportGroupBy.builder()
+                  .dataType(ReportsJurorPaymentsDataTypes.ATTENDANCE_DATE)
+                  .removeGroupByFromResponse(true)
+                  .build(),
+            ReportsJurorPaymentsDataTypes.JUROR_NUMBER,
+            ReportsJurorPaymentsDataTypes.FIRST_NAME,
+            ReportsJurorPaymentsDataTypes.LAST_NAME,
+            ReportsJurorPaymentsDataTypes.CHECKED_IN,
+            ReportsJurorPaymentsDataTypes.CHECKED_OUT,
+            ReportsJurorPaymentsDataTypes.HOURS_ATTENDED,
+            ReportsJurorPaymentsDataTypes.ATTENDANCE_AUDIT,
+            ReportsJurorPaymentsDataTypes.PAYMENT_AUDIT,
+            ReportsJurorPaymentsDataTypes.TOTAL_DUE,
+            ReportsJurorPaymentsDataTypes.TOTAL_PAID);
+
+        this.trialRepository = trialRepository;
+
+        isCourtUserOnly();
+    }
+
+    @Override
+    public Class<? extends Validators.AbstractRequestValidator> getRequestValidatorClass() {
+        return RequestValidator.class;
+    }
+
+    @Override
+    protected void preProcessQuery(JPAQuery<Tuple> query, StandardReportRequest request) {
+        query.where(QReportsJurorPayments.reportsJurorPayments.trialNumber.eq(request.getTrialNumber()));
+        query.orderBy(QReportsJurorPayments.reportsJurorPayments.jurorNumber.asc());
```

Comment:
Need to filter by loc code

Reply:
Added locCode check and integration test